packages/@aws-cdk/aws-ecs/lib/base/base-service.ts

```diff
@@ -356,6 +369,13 @@ export abstract class BaseService extends Resource
 
     this.taskDefinition = taskDefinition;
 
+    // launchType will set to undefined if using external DeploymentController or capacityProviderStrategies (whether
+    // specified on service itself or as default strategy on cluster)
+    const launchType = props.deploymentController?.type === DeploymentControllerType.EXTERNAL ||
```

Comment:
Can we add validation logic that throws an exception if they have both launchtype and capacity provider strategy defined? Using one silently might confuse users as to what they could expect.

Reply:
So my thought was that since the customers never actually pass in the LaunchType themselves (it's set automatically in the constructor depending on whether it's an Ec2Service or FargateService) that we should continue to handle the launchType setting for them under the hood. I saw that @pahud had thrown an error in his [PR](https://github.com/aws/aws-cdk/pull/12822/files#diff-becce6466790cd3cc81807ab64dd5f4ef85eed0285509d1ae43b381f24aefddaR377) but I felt like that would be less user friendly, again since the customer doesn't ever pass in launchType directly.

Reply:
See updated design doc for handling of launchType.